runtime-sdk/src/crypto/random.rs

```diff
@@ -0,0 +1,49 @@
+use rand_chacha::ChaChaRng;
+use rand_core::{CryptoRng, RngCore, SeedableRng as _};
+
+use crate::context::Context;
+
+pub struct Rng(ChaChaRng);
+
+impl Rng {
+    /// Create a new CSPRNG using a unique `nonce` and (optional, but useful) personalization string.
+    pub(crate) fn new<C>(ctx: &C, nonce: u64, pers: &[&[u8]]) -> Option<Self>
+    where
+        C: Context + ?Sized,
+    {
+        let km = ctx.key_manager()?;
+        let nonce_bytes = nonce.to_be_bytes();
+        let round_header_hash = ctx.runtime_header().encoded_hash();
+        let ctx_pers = [
+            b"oasis-runtime-sdk/crypto: random_bytes".as_slice(),
+            nonce_bytes.as_slice(),
+            &[ctx.mode() as u8],
+            round_header_hash.as_ref(),
+        ];
+        let key_id =
+            crate::keymanager::get_key_pair_id(ctx_pers.into_iter().chain(pers.iter().copied()));
+        let keypair = km.get_or_create_keys(key_id).ok()?;
```

Comment:
I know there's domain separation and everything, but I would rather you didn't do this.

In an ideal (everyone has lots of time) world, the key manager could implement a real entropy source for these sorts of requests that's distinct from the master secret used to derive actual keys.

Alternatively, something like this is less... "abusive".
```
sekrit = keymanager.get_keypair_id(my_contract_key)
pi = ECVRF_prove(sekrit, h(pers | round_header_hash | nonce)
beta = ECVRF_proof_to_hash(pi) // <- Use this as the seed
```

Reply:
Thanks for the review. I updated it to use the [`schnorkell` VRF](https://docs.rs/schnorrkel/0.10.2/schnorrkel/vrf/index.html) since the signature scheme seems already to be being used. Please let me know if this is a better use of primitives.

I also went and removed all of the user-provided personalization because it shouldn't be necessary as long as there's only one `Rng` created per block (and if I'm going to pretend to be a 3rdparty dev, I'm going to set it to `&[]` every time).

Reply:
Sure, it's a bit non-standard, but I have nothing against sr25519.